### ISSUE TITLE
Enumerator#any? is wrong for sequence of empty arrays

### ISSUE THREAD

**Issue by buehmann** [jruby/jruby#5968]
### Environment

```console
$ cat jruby_bug.rb 
p [[]].any?
p [[]].to_enum.any?
```

### Expected Behavior

```console
$ ruby -v                      
jruby 9.2.8.0 (2.5.3) 2019-08-12 a1ac7ff OpenJDK 64-Bit Server VM 11.0.5-ea+10-post-Ubuntu-0ubuntu1 on 11.0.5-ea+10-post-Ubuntu-0ubuntu1 +jit [linux-x86_64]
$ ruby jruby_bug.rb 2>/dev/null
true
true
```
(same for MRI)

### Actual Behavior

```console
$ ruby -v                      
jruby 9.2.9.0 (2.5.7) 2019-10-30 458ad3e OpenJDK 64-Bit Server VM 11.0.5-ea+10-post-Ubuntu-0ubuntu1 on 11.0.5-ea+10-post-Ubuntu-0ubuntu1 +jit [linux-x86_64]
$ ruby jruby_bug.rb 2>/dev/null
true
false
```

Discovered through https://github.com/rubocop-hq/rubocop/pull/7489

**Comment by ahorek** [comment]
bisected to this change https://github.com/jruby/jruby/commit/d16918b13d6c1c7cf19a578c73f2f9b9ab84837a

here's a different example that better explains the behavior

actual
```
 [[]].to_enum.take(1)
=> [nil]
```

expected
```
 [[]].to_enum.take(1)
=> [[]]
```

cc @kares

**Comment by kares** [comment]
oh nice, so we got a regression - seems like the test coverage isn't perfect.

was basically trying to increase compatibility (there's some more commits involved with Enumerator work in 9.2.9 than the one @ahorek bisected for us) and fix some long standing bugs.

hopefully we can fix this (pathological) case without regressing other stuff we started passing in 9.2.9

**Comment by koic** [comment]
Hi @kares. This issue doesn't seem to be resolved in JRuby 9.2.10.0 and 9.2.11.0.

```ruby
% cat jruby_bug.rb
p [[]].any?
p [[]].to_enum.any?
```

## JRuby 9.2.8.0

```console
% ruby -v
jruby 9.2.8.0 (2.5.3) 2019-08-12 a1ac7ff Java HotSpot(TM) 64-Bit Server VM 25.5-b02 on 1.8.0_05-b13 +jit [darwin-x86_64]
% ruby jruby_bug.rb
true
true
```


## JRuby 9.2.10.0

```console
% ruby -v
jruby 9.2.10.0 (2.5.7) 2020-02-18 fffffff Java HotSpot(TM) 64-Bit Server VM 25.5-b02 on 1.8.0_05-b13 +jit [darwin-x86_64]
% ruby jruby_bug.rb
true
false
```

## JRuby 9.2.11.0

```console
% ruby -v
jruby 9.2.11.0 (2.5.7) 2020-03-02 612d7a05a6 Java HotSpot(TM) 64-Bit Server VM 25.5-b02 on 1.8.0_05-b13 +jit [darwin-x86_64]
% ruby jruby_bug.rb
true
false
```

**Comment by headius** [comment]
Retargeted to 9.2.12. See #6285.
